src/core/renderer.js

```diff
@@ -577,7 +578,27 @@ Renderer.prototype.refresh = function (timestamp) {
 
 
     const s = 1. / this._zoom;
-    tiles.forEach(tile => {
+
+    // TODO remove hack
+    let orderer = null;
```

Comment:
🌈  For a newcomer with no context like me this code fragment looks like an ancient Egyptian writing. Of course it is my responsibility to know the code base and I don't (yet) but some improvements could me made in order to improve understandability.

the `Renderer.refresh` method is quite big I think if we [extract functions](https://refactoring.com/catalog/extractMethod.html) and give those functions a good descriptive name the code readability will improve a lot.

Reply:
I really liked the "ancient Egyptian writing", but having long functions with WebGL/OpenGL code is not frowned upon by many OpenGL developers because it actually helps to reduce errors. This is basically because if you extract a method, you need to explain quite well how that method is going to change WebGL state, and what kind of WebGL state it does expect. This is extremely error-prone. 

I recommend seeing Jonathan Blow talks, particularly this one that talks about this issue: https://youtu.be/JjDsP5n2kSM?t=27m40s

Reply:
I could agree when the new created function has side effects but in this case we are talking of a `pure function` so there is no risk of changing webgl state.

```javascript
function goodFunctionName(tiles) {
    let orderer = null;
    let orderingMins = null;
    let orderingMaxs = null;

    if (tiles.length > 0) {
        orderer = tiles[0].style._order;
    }
    if (orderer instanceof ordering.Asc) {
        orderingMins = Array.from({ length: 16 }, (_, i) => (15 - i) * 2);
        orderingMaxs = Array.from({ length: 16 }, (_, i) => i == 0 ? 1000 : (15 - i + 1) * 2);
    }
    else if (orderer instanceof ordering.Desc) {
        orderingMins = Array.from({ length: 16 }, (_, i) => i * 2);
        orderingMaxs = Array.from({ length: 16 }, (_, i) => i == 15 ? 1000 : (i + 1) * 2);
    }
    else {
        orderingMins = [0];
        orderingMaxs = [1000];
    }
    return { orderingMins, orderingMaxs };
}
```

And in the renderer.refresh:

```javascript
const { orderingMins, orderingMaxs } = getOrdererMinandMax(tiles);
```